mu4/notation/view/notationpaintview.cpp

```diff
@@ -360,19 +346,22 @@ void NotationPaintView::dropEvent(QDropEvent* ev)
     m_inputController->dropEvent(ev);
 }
 
-QPoint NotationPaintView::toLogical(const QPoint& p) const
+QPoint NotationPaintView::toLogical(const QPoint& point) const
 {
-    return m_matrix.inverted().map(p);
+    double scale = guiScale();
+    QPoint scaledPoint(point.x() * scale, point.y() * scale);
+
+    return m_matrix.inverted().map(scaledPoint);
 }
 
-QRect NotationPaintView::toLogical(const QRect& r) const
+double NotationPaintView::guiScale() const
 {
-    return m_matrix.inverted().mapRect(r);
+    return window()->screen()->devicePixelRatio();
```

Comment:
we have `UiConfiguration::guiScaling`, I think better use it (and move this code there)

Reply:
This is a problem. Here need to get the scale of the current screen (on which a view is placed). I could not find any solution without using window() of QQuickPaintedItem

Reply:
there is `IMainWindow` interface, which can be used in the UiConfiguration by injection

Reply:
fixed